pve2_api.class.php

```diff
@@ -251,20 +261,21 @@ private function action ($action_path, $http_method, $put_post_parameters = null
 					return $action_response_array['data'];
 				}
 			} else {
-				error_log("This API Request Failed.\n" .
-					"HTTP Response - {$split_http_response_line[1]}\n" .
-					"HTTP Error - {$split_headers[0]}");
+				throw new PVE2_Exception("PVE2 API: This API Request Failed.\n" .
+					"HTTP CODE: {$split_http_response_line[1]},\n" .
+					"HTTP ERROR: {$split_headers[0]},\n" . 
+					"REPLY INFO: {$body_response}");
 				return false;
 			}
 		} else {
-			error_log("Error - Invalid HTTP Response.\n" . var_export($split_headers, true));
+			throw new PVE2_Exception("PVE2 API: Error - Invalid HTTP Response.\n" . var_export($split_headers, true));
 			return false;
```

Comment:
Throw/return paradox again.

Reply:
Not sure if you want everything reverted to error_log. Excepting is helpful as we can expect a proper reply in integrated apps, and relay the error specifics to the end user. So I'm hesitant to revert everything?

Reply:
All I meant was that once you `throw`, the `return` never gets called.